nix/default.nix

```diff
@@ -1,45 +1,49 @@
 { system ? builtins.currentSystem }:
 let
-  pkgs = import ./nixpkgs.nix {};
+  pkgs = (import ./nixpkgs.nix {
+    config = {
+      packageOverrides = pkg: {
+        gpgme = (static pkg.gpgme);
+        libassuan = (static pkg.libassuan);
+        libgpgerror = (static pkg.libgpgerror);
+        libseccomp = (static pkg.libseccomp);
+      };
+    };
+  });
 
-  static = pkg: pkg.overrideAttrs(old: {
-    configureFlags = (old.configureFlags or []) ++
+  static = pkg: pkg.overrideAttrs(x: {
+    configureFlags = (x.configureFlags or []) ++
       [ "--without-shared" "--disable-shared" ];
     dontDisableStatic = true;
     enableSharedExecutables = false;
     enableStatic = true;
   });
 
-  patchLvm2 = pkg: pkg.overrideAttrs(old: {
-    configureFlags = [
-      "--disable-cmdlib" "--disable-readline" "--disable-udev_rules"
-      "--disable-udev_sync" "--enable-pkgconfig" "--enable-static_link"
-    ];
-    preConfigure = old.preConfigure + ''
-      substituteInPlace libdm/Makefile.in --replace \
-        SUBDIRS=dm-tools SUBDIRS=
-      substituteInPlace tools/Makefile.in --replace \
-        "TARGETS += lvm.static" ""
-      substituteInPlace tools/Makefile.in --replace \
-        "INSTALL_LVM_TARGETS += install_tools_static" ""
+  self = with pkgs; buildGoPackage rec {
```

Comment:
I would like to stick to buildGoModule (see latest nixpkgs)

Reply:
I tried that once but result not good:
- The result file `./result/bin/crio` become a shell script file, which wrap the initialization of environment variables plus running the target binary
- The target binary still not pointing to the correct version. This happened once when I upgrade the nixpkgs.json into latest for podman, which always pointing to v1.9.3 as nixpkgs upstream

I found that the key successful factor when we pinned with the legacy archive is: it was using buildGoPackage but not the latest buildGoModule.

That's also why I prefer to clone the legacy successful structure but not override from the latest: latest upstream's structure for dynamically linked may make us fail unexpectedly for building statically linked binary. So let's DIY ;-)

Reply:
Revamp with `buildGoModule`